ehrapy/preprocessing/_imputation.py

```diff
@@ -274,40 +246,26 @@ def knn_impute(
         from sklearnex import patch_sklearn, unpatch_sklearn
 
         patch_sklearn()
+
     try:
-        with Progress(
-            "[progress.description]{task.description}",
-            SpinnerColumn(),
-            refresh_per_second=1500,
-        ) as progress:
-            progress.add_task("[blue]Running KNN imputation", total=1)
-            # numerical only data needs no encoding since KNN Imputation can be applied directly
-            if np.issubdtype(adata.X.dtype, np.number):
-                _knn_impute(adata, var_names, n_neighbors, backend=backend, **backend_kwargs)
-            else:
-                # ordinal encoding is used since non-numerical data can not be imputed using KNN Imputation
-                enc = OrdinalEncoder()
-                column_indices = adata.var[FEATURE_TYPE_KEY] == CATEGORICAL_TAG
-                adata.X[::, column_indices] = enc.fit_transform(adata.X[::, column_indices])
-                # impute the data using KNN imputation
-                _knn_impute(adata, var_names, n_neighbors, backend=backend, **backend_kwargs)
-                # imputing on encoded columns might result in float numbers; those can not be decoded
-                # cast them to int to ensure they can be decoded
-                adata.X[::, column_indices] = np.rint(adata.X[::, column_indices]).astype(int)
-                # knn imputer transforms X dtype to numerical (encoded), but object is needed for decoding
-                adata.X = adata.X.astype("object")
-                # decode ordinal encoding to obtain imputed original data
-                adata.X[::, column_indices] = enc.inverse_transform(adata.X[::, column_indices])
+        if np.issubdtype(adata.X.dtype, np.number):
+            _knn_impute(adata, var_names, n_neighbors, backend=backend, **backend_kwargs)
+        else:
+            # Raise exception since non-numerical data can not be imputed using KNN Imputation
+            raise ValueError(
+                "Can only impute numerical data. Try to restrict imputation to certain columns using "
+                "var_names parameter or."
```

Comment:
alternatively to be in sync with miss_forest and mice_forest:
```suggestion
                "var_names parameter."
```